numpy/core/__init__.py

```diff
@@ -126,14 +125,17 @@ def _ufunc_reconstruct(module, name):
     return getattr(mod, name)
 
 def _ufunc_reduce(func):
-    from pickle import whichmodule
-    name = func.__name__
-    return _ufunc_reconstruct, (whichmodule(func, name), name)
+    # Report the `__name__`. pickle will try to find the module. Note that
+    # pickle supports for this `__name__` to be a `__qualname__`. It may
+    # make sense to add a `__qualname__` to ufuncs, to allow this more
+    # explicitly (Numba has ufuncs as attributes).
+    # See also: https://github.com/dask/distributed/issues/3450
+    return func.__name__
```

Comment:
Is this behavior of returning strings documented anywhere? I don't see it in https://docs.python.org/3/library/pickle.html.

Reply:
Its a bit confusingly documented in [`copyreg`](https://docs.python.org/3/library/copyreg.html#copyreg.pickle)

There is also a note on the [`__reduce__` doctsring](https://docs.python.org/3/library/pickle.html#object.__reduce__) (second paragraph)

I would be happy to just implement `__reduce__` as well though, I guess I felt this is a start on that route.